### ISSUE TITLE
Can you cast a big integer literal?

### ISSUE THREAD

**Issue by BryantLam** [chapel-lang/chapel#11500]
Reusing parent code from #11381 with modification on `y` to be a cast operation:
```chapel
var x = 0x812341234:uint(32); // Cast value to uint(32)
writef("%xu\n", x);

var y = 0x81234123412341234:uint(64); // Cast big integer to uint(64)
writef("%xu\n", y);
```
emits
```
error: Integer literal overflow: '0x81234123412341234' is too big for type uint64
```

This is in contrast to var `x` that succeeds on a cast because the literal could fit the uint64 type before the cast happens. I don't think I care what the resolution of this issue is, but it does seem like it could impact BigInteger `init=` if you can't actually assign it a big integer (because of type limit on uint64), or if you could someday do something akin to `var z = <big integer literal>:BigInteger;`

**Comment by cassella** [comment]
#11397 added some documentation about this to BigInteger.chpl.

**Comment by mppf** [comment]
Right, in other words, big integer literals don't exist right now in Chapel.
